dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Extensions/AnalysisContextExtensions.cs

```diff
@@ -19,23 +20,24 @@ internal static void SkipEmptyName(this SymbolAnalysisContext context, Action<Sy
             }
         }
 
-        internal static void SkipEmptyName(this SyntaxNodeAnalysisContext context, Action<SymbolAnalysisContext> action)
+        internal static void SkipEmptyName(
+            this SyntaxNodeAnalysisContext context,
+            Action<SyntaxNodeAnalysisContext> action)
         {
-            SymbolAnalysisContext symbolContext = context.ToSymbolContext();
-            symbolContext.SkipEmptyName(_ => action(symbolContext));
-        }
-
-        internal static SymbolAnalysisContext ToSymbolContext(this SyntaxNodeAnalysisContext syntaxContext)
-        {
-            ISymbol symbol = syntaxContext.SemanticModel.GetDeclaredSymbol(syntaxContext.Node);
-
-            return SyntaxToSymbolContext(syntaxContext, symbol);
+            if (context.Node is ParameterSyntax parameterSyntax &&
+                !string.IsNullOrWhiteSpace(parameterSyntax.Identifier.ValueText))
+            {
+                action(context);
+            }
         }
 
-        private static SymbolAnalysisContext SyntaxToSymbolContext(SyntaxNodeAnalysisContext syntaxContext, ISymbol symbol)
+        private static CompilationWithAnalyzers SyntaxToSymbolContext(SyntaxNodeAnalysisContext syntaxContext,
+            ISymbol symbol)
         {
-            return new SymbolAnalysisContext(symbol, syntaxContext.SemanticModel.Compilation, syntaxContext.Options,
```

Comment:
@richardb355 I got a build warning about this constructor being deprecated, so needed to keep it as a `SyntaxNodeAnalysisContext` and update the abbreviations analyzer to handle it

Reply:
Happy to resolve this.